app/controllers/orders_controller.rb

```diff
@@ -114,6 +114,24 @@ def cancel
     redirect_to checkout_order_path, alert: 'Payment was cancelled'
   end
 
+  def cancel_order
+    if @order.deals.any? { |deal| deal.expiring_soon? }
+      flash[:notice] = "Order can't be cancelled 30 minutes before the deal ends"
+    else
+      payment_intent = Stripe::PaymentIntent.retrieve(@order.payments.successful.payment_intent)
+      refund = Stripe::Refund.create({
+        payment_intent: payment_intent
+      })
+      @order.refunds.create(refund_id: refund.id, status: 'Successful', currency: 'inr', total_amount_in_cents: refund.amount)
```

Comment:
why is it refund.amount and not refund.amount_in_cents

Reply:
because refund is an object of Stripe::Refund, and `amount` is one of its attribute.